pkg/network/setup/netpod/netpod.go

```diff
@@ -210,22 +238,30 @@ func (n NetPod) composeDesiredSpec(currentStatus *nmstate.Status) (*nmstate.Spec
 		)
 		podIfaceName := podIfaceNameByVMINetwork[iface.Name]
 
-		// Filter out network interfaces marked for removal.
-		// TODO: Support in the same flow the removal of such interfaces.
-		if iface.State == v1.InterfaceStateAbsent {
-			continue
-		}
-
 		switch {
 		case iface.Bridge != nil:
-			if _, exists := podIfaceStatusByName[podIfaceName]; !exists {
+			// A missing pod interface is not considered an error in case the interface is marked for removal.
+			if _, exists := podIfaceStatusByName[podIfaceName]; !exists && iface.State != v1.InterfaceStateAbsent {
 				return nil, fmt.Errorf("pod link (%s) is missing", podIfaceName)
 			}
 			ifacesSpec, err = n.bridgeBindingSpec(podIfaceName, ifIndex, podIfaceStatusByName)
 
 			if nmstate.AnyInterface(ifacesSpec, hasIP4GlobalUnicast) {
 				spec.LinuxStack.IPv4.ArpIgnore = pointer.P(procsys.ARPReplyMode1)
 			}
+
+			if iface.State == v1.InterfaceStateAbsent {
+				var filteredIfacesSpec []nmstate.Interface
+				for _, ifaceSpec := range ifacesSpec {
+					// Interfaces with no type are not owned by kubevirt, therefore not removed.
+					if ifaceSpec.TypeName != "" {
```

Comment:
Eventually only interfaces that specified in the VMI spec are processed, regardless of its type, isnt it?